fuel-core-interfaces/src/model/block.rs

```diff
@@ -598,60 +705,19 @@ impl SealedFuelBlock {
     }
 }
 
-impl From<FuelBlock> for PartialFuelBlock {
-    fn from(block: FuelBlock) -> Self {
-        let FuelBlock {
-            header:
-                FuelBlockHeader {
-                    application: FuelApplicationHeader { da_height, .. },
-                    consensus:
-                        FuelConsensusHeader {
-                            prev_root,
-                            height,
-                            time,
-                            ..
-                        },
-                    ..
-                },
-            transactions,
-        } = block;
-        Self {
-            header: PartialFuelBlockHeader {
-                application: FuelApplicationHeader {
-                    da_height,
-                    generated: Empty {},
-                },
-                consensus: FuelConsensusHeader {
-                    prev_root,
-                    height,
-                    time,
-                    generated: Empty {},
-                },
-                metadata: None,
-            },
-            transactions,
-        }
-    }
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(any(test, feature = "test-helpers"), derive(Default))]
+/// A fuel block with the related consensus data.
+pub struct SealedFuelBlockHeader {
+    pub header: FuelBlockHeader,
+    pub consensus: FuelBlockConsensus,
 }
 
-#[cfg(any(test, feature = "test-helpers"))]
-impl<T> Default for FuelConsensusHeader<T>
-where
-    T: Default,
-{
-    fn default() -> Self {
-        Self {
-            time: Utc.timestamp(0, 0),
-            height: BlockHeight::default(),
-            prev_root: Bytes32::default(),
-            generated: Default::default(),
-        }
-    }
-}
+impl Deref for SealedFuelBlockHeader {
```

Comment:
I would think that providing deref behaviour would mean using the type as a pointer, so you can use pointer dereferencing syntax, e.g. 
```rust
...
let value = *v;
```
Is `SealedFuelBlockHeader` being used as a pointer type? 
From an API standpoint, this adds an implicit deref operation that returns the header; are there any design concerns there?

Reply:
We probably abuse deref in many places to convert types when we probably should be using conversion traits like AsRef instead or simply use getters. Probably due to wanting concise `*` syntax.

I'll look into an alternative approach.

Reply:
removed deref

Reply:
Yeh technically we should probably be using `as_ref()` but it's more of a pain.

Reply:
Interesting read! Thanks for the context. It appears to be a convenient pattern but I would agree with the document you posted that it is a surprising mechanism. But it's good to know how and why this comes about.